scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -247,15 +241,27 @@
   cc/ComputeCluster
   (launch-tasks [this offers task-metadata-seq]
     (doseq [task-metadata task-metadata-seq]
-      (let [pod-namespace (get-namespace-from-task-metadata namespace-config task-metadata)]
+      ; Has the workload of launching tasks changed or has the cost of doing them changed?
+      ; Note we can't use timer/time! because it wraps the body in a Callable, which rebinds 'this' to another 'this'
+      ; causing breakage.
+      ; In addition, it capture controller/update-cook-expected-state and
+      ; screw up when we do a with-redefs on it in unit tests.
+      (let [timer-context (timers/start (metrics/timer "compute-cluster-launch-tasks" name))
+            pod-namespace (get-namespace-from-task-metadata namespace-config task-metadata)]
         (controller/update-cook-expected-state
           this
           (:task-id task-metadata)
           {:cook-expected-state :cook-expected-state/starting
-           :launch-pod {:pod (api/task-metadata->pod pod-namespace name task-metadata)}}))))
+           :launch-pod {:pod (api/task-metadata->pod pod-namespace name task-metadata)}})
+        (.stop timer-context))))
 
   (kill-task [this task-id]
-    (controller/update-cook-expected-state this task-id {:cook-expected-state :cook-expected-state/killed}))
+    ; Note we can't use timer/time! because it wraps the body in a Callable, which rebinds 'this' to another 'this'
+    ; causing breakage.
+    (let [timer-context (timers/start (metrics/timer "compute-cluster-kill-tasks" name))]
+      ; Has the workload of killing tasks changed or has the cost of doing them changed?
```

Comment:
I would vote to delete this comment - I don't think we need to justify the metrics.

Reply:
Ok.